app/Components/SudokuBoard/Functions/HintFunctions.ts

```diff
@@ -0,0 +1,85 @@
+import { getHint, SudokuStrategy } from "sudokuru";
+import { CellProps } from "../../../Functions/LocalDatabase";
+import { Hint } from "../SudokuBoard";
+import { generateBoxIndex } from "../sudoku";
+
+export const getSudokuHint = (
+  puzzle: CellProps[][],
+  solution: number[][],
+  strategies: SudokuStrategy[]
+): Hint => {
+  const puzzleState = convertPuzzleStateToSudokuruFormat(puzzle);
+  const puzzleSolution = convertPuzzleSolutionToSudokuruFormat(solution);
+  let hint = getHint(
+    puzzleState.puzzleValues,
+    puzzleState.puzzleNotes,
+    strategies,
+    puzzleSolution
+  ) as unknown as Hint;
```

Comment:
curious why unknown?

Reply:
That was the only way I was able to assign it the Hint type when I implemented it

Reply:
This was the only way I could get it to work. With the sudokuru package rewrite I would prefer to have the types exported from the package and we would not have this issue.